pageserver/src/tenant/mgr.rs

```diff
@@ -151,61 +151,86 @@ async fn safe_rename_tenant_dir(path: impl AsRef<Utf8Path>) -> std::io::Result<U
 
 static TENANTS: Lazy<RwLock<TenantsMap>> = Lazy::new(|| RwLock::new(TenantsMap::Initializing));
 
-/// Initialize repositories with locally available timelines.
-/// Timelines that are only partially available locally (remote storage has more data than this pageserver)
-/// are scheduled for download and added to the tenant once download is completed.
-#[instrument(skip_all)]
-pub async fn init_tenant_mgr(
-    conf: &'static PageServerConf,
-    resources: TenantSharedResources,
-    init_order: InitializationOrder,
-    cancel: CancellationToken,
-) -> anyhow::Result<()> {
-    // Scan local filesystem for attached tenants
-    let tenants_dir = conf.tenants_path();
+fn emergency_generations(
+    tenant_confs: &HashMap<TenantId, anyhow::Result<LocationConf>>,
+) -> HashMap<TenantId, Generation> {
+    tenant_confs
+        .iter()
+        .filter_map(|(tid, lc)| {
+            let lc = match lc {
+                Ok(lc) => lc,
+                Err(_) => return None,
+            };
```

Comment:
```suggestion
            let lc = lc.ok()?;
```

Though, suggesting this while really thinking, has someone logged this error already?

Reply:
Oki found where this error is logged normally, perhaps makes sense not to have these errors in emergencies.

If there should never be these errors then I guess it could be a `expect("should never exist in emergencies")`, if that is statically so.

Reply:
It's an ugly corner: these errors are never expected, but in the event it happens, we would like to reduce the blast radius to the tenant in question, rather than refusing to proceed.  So as you noticed, we just log the error later when we refuse to spin up a Tenant because of no config.